### PR TITLE
Refactor content method

### PR DESCRIPTION
**Pull request by jonathanyeong**
Break down the content method into smaller pieces. Also tried to move any repeating logic into it's own method. @sarony.

### CODE REVIEW

File: lib/html_factory.rb [hunk 1]
```diff
@@ -675,29 +675,21 @@ def body_end
     DOC
   end
     
-  def content(content_lists, callouts, sponsors)
+  def content(newsletter_content)
+    string = ""
+    string << content_list_content(newsletter_content[:content_lists])
+    string << sponsor_content(newsletter_content[:sponsors])
+    string << callout_content(newsletter_content[:callouts])
+  end
```
Comment (sarony) [inline lib/html_factory.rb:683]: This looks SO MUCH BETTER. How do you feel about it?

File: spec/views/index.html_spec.rb [hunk 2]
```diff
@@ -23,5 +23,7 @@
   it "contains newsletter content" do
     expect(@page.text).to include("Podcast")
     expect(@page.text).to include("ICYMI")
+    expect(@page.text).to include("callout")
+    expect(@page.text).to include("Sponsors")
```
Comment (sarony) [inline spec/views/index.html_spec.rb:27]: Let's add specs that correspond to the breakout methods.

File: lib/html_factory.rb [hunk 3]
```diff
@@ -709,39 +701,26 @@ def content(content_lists, callouts, sponsors)
...
-      stripped_post = post.body.gsub("<p>","").gsub("</p>","")
-      content_string = <<-DOC.gsub(/^ {4}/, '')
-        <div class="clearfix" style="clear:both;">
-            <a href="#{post.link}"><h2 class="h2">#{post.title}</h2></a>
-            <p class="photo-content">#{stripped_post}</p>
-            <a href="#{post.link}" target="_blank"><img class="photo" src="#{post.attachment}" alt="Blog guest picture"></a>
-       </div>
-      DOC
-      string << content_string 
+      string << picture_post(post)
```
Comment (sarony) [inline lib/html_factory.rb:714]: very nice

File: lib/html_factory.rb [hunk 4]
```diff
@@ -760,4 +739,33 @@ def content(content_lists, callouts, sponsors)
     DOC
     string << content_string
   end
+
+  def section_title(html_class, title, has_hr)
+    if has_hr
```
Comment (sarony) [inline lib/html_factory.rb:744]: is "Join Us" the only one with no hr? If so, it might make more sense to have a default value of true.

**Comment by sarony** [comment]
@jonathanyeong made some minor comments. Once you take a look and respond, ping me and I'll take another look.

File: lib/html_factory.rb (hunk 1 above)
Comment (jonathanyeong) [inline lib/html_factory.rb:683]: It feels so good! That beautiful 4 line method.

File: lib/html_factory.rb (hunk 4 above)
Comment (jonathanyeong) [inline lib/html_factory.rb:744]: Yeah it is. That's a good idea. I'll make the change.

File: spec/views/index.html_spec.rb (hunk 2 above)
Comment (jonathanyeong) [inline spec/views/index.html_spec.rb:27]: Okay will do.

**Pull request merged**